src/shared/remote/remoteSourceTrackingService.ts

```diff
@@ -112,13 +120,14 @@ export class RemoteSourceTrackingService {
    */
   public static async getInstance(options: RemoteSourceTrackingServiceOptions): Promise<RemoteSourceTrackingService> {
     const orgId = options.org.getOrgId();
-    let service = this.instanceMap.get(orgId);
-    if (!service) {
-      service = await new RemoteSourceTrackingService(options).init();
-      this.instanceMap.set(orgId, service);
-    }
+    const service = this.instanceMap.get(orgId)?.service ?? (await new RemoteSourceTrackingService(options).init());
+    this.instanceMap.set(orgId, { service, lastUsed: Date.now() });
+    // when we create a new instances, we make sure old ones are not accumulating.  Important in multitenant environments
```

Comment:
This comment doesn't seem accurate.  Looks like whenever `RSTS.getInstance()` is called it will purge, not when a new instance is created.